packages/components/src/components/linear-progress/linear-progress-container.jsx

```diff
@@ -2,20 +2,41 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { LinearProgress } from './linear-progress.jsx';
 
-const LinearProgressContainer = ({ timeout, action, render, className }) => {
-    const [timeout_state, setTimeout] = React.useState(timeout / 1000);
-    const [current_tick, setCurrentTick] = React.useState(Math.round(timeout / 1000));
-    const [total_ticks] = React.useState(Math.round(timeout / 1000));
+const LinearProgressContainer = ({
+    timeout,
+    action,
+    render,
+    className,
+    should_store_in_session,
+    session_id,
+    linear_progress_container_ref,
+}) => {
+    const current_progress_timeout = sessionStorage.getItem(`linear_progress_timeout_${session_id}`);
 
-    const getProgress = () => 100 - Math.round((current_tick / total_ticks) * 100);
+    const popup_timeout = !current_progress_timeout ? timeout / 1000 : current_progress_timeout;
+    const [timeout_state, setTimeoutState] = React.useState(popup_timeout);
+    const time_past = 100 - timeout_state / 3;
```

Comment:
why are we dividing by 3?

Reply:
The timeout bar goes from 0 - 100. The timeout_state is 300 seconds and has to be divided by 3 so that 100 - 100 results in 0 (the start point of the progress bar). Then it can count up 300 ticks.

for example:

0 [///////__________] 100

0.3... 0.6... 0.9... 1.2.... etc. (till it reaches 100, then the bar is filled and the popup refreshes the browser).

Reply:
Ah okay. The timeout is passed as a prop to the component. So, the `timeout` is dynamic and the calculation should be dynamic as well. If we change the timeout to 500 seconds, this won't work as expected.

The below line will make the code dynamic.
`
100 - (timeout_state / timeout * 100)
`

Reply:
Aaaahh I see, ofcourse. I applied your fix, Although, I slightly modified it: `const time_past = 100 - (timeout_state / (timeout / 1000) * 100)` Since timeout_state is already divided by 1000.